tasks/GroundFollower.cpp

```diff
@@ -0,0 +1,146 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "GroundFollower.hpp"
+//#include <math.h>
+
+using namespace auv_control;
+
+GroundFollower::GroundFollower(std::string const& name, TaskCore::TaskState initial_state)
+    : GroundFollowerBase(name, initial_state)
+{
+}
+
+GroundFollower::GroundFollower(std::string const& name, RTT::ExecutionEngine* engine, TaskCore::TaskState initial_state)
+    : GroundFollowerBase(name, engine, initial_state)
+{
+}
+
+GroundFollower::~GroundFollower()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See GroundFollower.hpp for more detailed
+// documentation about them.
+
+bool GroundFollower::configureHook()
+{
+    if (! GroundFollowerBase::configureHook())
+        return false;
+
+    new_altimeter_timeout = base::Timeout(_altimeter_timeout.get());
+    new_depth_timeout = base::Timeout(_depth_timeout.get());
+    altimeter_dropout_timeout = base::Timeout(_altimeter_dropout_timeout.get());
+
+    distance_to_ground_cmd = _distance_to_ground.get();
+    if(distance_to_ground_cmd <= 0){
+        exception(INVALID_TARGET_DEPTH_CONFIG);
```

Comment:
I'm not sure if going to an exception state from the configuration hook is valid.
Returning false should be sufficient to signal that the configuration failed.

Reply:
hmm yeah, since we are returning false, it will bring the task to a pre-operational state. But what would be a way to tell the user that there's a wrong config in this case?

Reply:
You could write out a message on the error stream of a logger.